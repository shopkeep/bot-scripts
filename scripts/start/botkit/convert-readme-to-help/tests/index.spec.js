const convertReadmeToHelp = require("../");
const readmeData = require("./test-data/readme-data");

jest.mock("fs");
const fs = require("fs");

describe("convertReadmeToHelp", function() {
  let testContext;

  beforeEach(function() {
    jest.clearAllMocks();
    testContext = {};
    testContext.conversionPromise = convertReadmeToHelp("./", {
      Goodbye: ":point_left:"
    });
  });

  it("pulls in the readme", function() {
    expect(fs.readFile.mock.calls.length).toBe(1);
  });

  describe("when unable to get readme", function() {
    beforeEach(function() {
      fs.readFile.mock.calls[0][2](new Error("test error"));
    });

    it("provides an error", async function() {
      expect.assertions(1);

      try {
        await testContext.conversionPromise;
      } catch (e) {
        expect(e).toEqual(new Error("test error"));
      }
    });
  });

  describe("when able to get readme", function() {
    describe("but unable to parse readme", function() {
      beforeEach(function() {
        fs.readFile.mock.calls[0][2](null, "hello world");
      });

      it("provides an error message", async function() {
        expect.assertions(1);

        try {
          await testContext.conversionPromise;
        } catch (e) {
          expect(e).toEqual(new Error("Incompatible readme structure"));
        }
      });
    });

    describe("and able to parse readme", function() {
      beforeEach(function() {
        fs.readFile.mock.calls[0][2](null, readmeData);
      });

      it("provides an help message", async function() {
        expect(await testContext.conversionPromise).toMatchSnapshot();
      });
    });
  });
});
