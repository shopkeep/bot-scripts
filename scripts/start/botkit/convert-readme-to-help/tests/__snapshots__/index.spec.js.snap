// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertReadmeToHelp when able to get readme and able to parse readme provides an help message 1`] = `
"Here's a list of available commands :information_desk_person: 

:point_right: _*Hello*_
Some details about hello command

:point_left: _*Goodbye*_
Some details bout the goodbye command"
`;
